src/providers/FileSystemProvider/FileSystemProvider.ts

```diff
@@ -699,6 +696,9 @@ export class FileSystemProvider implements vscode.FileSystemProvider {
     if (vscode.workspace.getWorkspaceFolder(oldUri) != vscode.workspace.getWorkspaceFolder(newUri)) {
       throw vscode.FileSystemError.NoPermissions("Cannot rename a file across workspace folders");
     }
+    if (oldUri.path.includes(".vscode/") || newUri.path.includes(".vscode/")) {
```

Comment:
VS Code will never do this, so it's purely precautionary.

Reply:
Please either remove it or add an explanatory comment to the code.